src/mixins/ERC4626.sol

```diff
@@ -120,33 +122,37 @@ abstract contract ERC4626 is ERC20 {
 
     function totalAssets() public view virtual returns (uint256);
 
-    function assetsOf(address user) public view virtual returns (uint256) {
-        return previewRedeem(balanceOf[user]);
+    function convertToShares(uint256 assets) public view returns (uint256 shares) {
+        uint256 supply = totalSupply; // Saves an extra SLOAD if totalSupply is non-zero.
+
+        return shares = supply == 0 ? assets : assets.mulDivUp(supply, totalAssets());
```

Comment:
this should mulDivDown not up to track the results of a deposit call imo. Open to tracking withdraw if thats preferred by @transmissions11

Reply:
I’d like to standardize this in the spec itself, can see arguments for both sides. Needs a bit more discussion

Reply:
oke we agreed to use mulDivDown

Reply:
I'll push the change, but just a heads up, this assertion won't hold : 

`vault.convertToShares(vault.convertToAssets(vault.balanceOf(x))) !== vault.balanceOf(x)`

because both functions round down (pessimisticly)

Reply:
![0B9AF509-3F9C-4554-8581-CA60C8D3A197](https://user-images.githubusercontent.com/26209401/156622084-aed3b761-e0cf-4111-ab78-f78d3ff3d51e.jpeg)

Even if we rounded up its still not reversible in all cases tho, no?

Reply:
I don't think so either